test/gst-vaapi/transcode/transcoder.py

```diff
@@ -11,45 +11,199 @@
+      ),
+      "hevc-8" : dict(
+        sw = (ALL_PLATFORMS, have_gst_element("avdec_h265"), ("avdec_h265", "h265parse")),
+        hw = (HEVC_DECODE_8BIT_PLATFORMS, have_gst_element("vaapih265dec"), ("vaapih265dec", "h265parse")),
+      ),
+      "mpeg2" : dict(
+        hw = (MPEG2_DECODE_PLATFORMS, have_gst_element("vaapimpeg2dec"), ("vaapimpeg2dec", "mpegvideoparse")),
+      ),
+      "mjpeg" : dict(
+        hw = (MPEG2_DECODE_PLATFORMS, have_gst_element("vaapijpegdec"), ("vaapijpegdec", "jpegparse")),
+      ),
+      "vc1" : dict(
+        hw = (VC1_DECODE_PLATFORMS, have_gst_element("vaapivc1dec"), ("vaapivc1dec", None)),
+      ),
+    },
+    encode = {
+      "avc" : dict(
+        sw = (ALL_PLATFORMS, have_gst_element("x264enc"), ("x264enc", "h264",  "video/x-h264,profile=main", "h264parse")),
+        hw = (AVC_ENCODE_PLATFORMS, have_gst_element("vaapih264enc"), ("vaapih264enc", "h264", "video/x-h264,profile=main", "h264parse", )),
+      ),
+      "hevc-8" : dict(
+        sw = (ALL_PLATFORMS, have_gst_element("x265enc"), ("x265enc", "h265", "video/x-h265,profile=main", "h265parse" )),
+        hw = (HEVC_ENCODE_8BIT_PLATFORMS, have_gst_element("vaapih265enc"), ("vaapih265enc", "h265", "video/x-h265,profile=main", "h265parse")),
+      ),
+      "mpeg2" : dict(
+        sw = (ALL_PLATFORMS, have_gst_element("avenc_mpeg2video"), ("avenc_mpeg2video", "m2v", None, "mpegvideoparse")),
+        hw = (MPEG2_ENCODE_PLATFORMS, have_gst_element("vaapimpeg2enc"), ("vaapimpeg2enc", "m2v", None, "mpegvideoparse")),
+      ),
+      "mjpeg" : dict(
+        sw = (ALL_PLATFORMS, have_gst_element("jpegenc"), ("jpegenc", "mjpeg", None, "jpegparse")),
+        hw = (JPEG_ENCODE_PLATFORMS, have_gst_element("vaapijpegenc"), ("vaapijpegenc", "mjpeg", None, "jpegparse")),
+      ),
+    }
+  )
+
+  # hevc implies hevc 8 bit
+  requirements["encode"]["hevc"] = requirements["encode"]["hevc-8"]
+  requirements["decode"]["hevc"] = requirements["decode"]["hevc-8"]
+
   def before(self):
     self.refctx = []
+  
+  def get_requirements_data(self, ttype, codec, mode):
+      plats, reqs, ffparams = self.requirements[ttype].get(
+        codec, {}).get(mode, ([], (False, False), None))
+      return plats, reqs, ffparams
 
-  def transcode_1to1(self):
-    self.transcoded = get_media()._test_artifact(
-      "{case}_{width}x{height}_{mode}.{dstextension}".format(**vars(self)))
-    get_media().test_call_timeout = vars(self).get("call_timeout", 0)
-    call("gst-launch-1.0 -vf filesrc location={source} ! {gsttrans}"
-      " ! filesink location={transcoded}".format(**vars(self)))
+  def validate_spec(self):
+    from slash.utils.pattern_matching import Matcher
 
-    self.convert_toyuv()
-    self.check_metrics()
+    assert len(self.outputs), "Invalid test case specification, outputs data empty"
+    assert self.mode in ["sw", "hw"], "Invalid test case specification as mode type not valid"
 
```

Comment:
I don't think gstreamer supports 1:n transcoding?  Thus, we need to skip the test if `len(self.outputs) > 1 or self.outputs[0].get("channels", 1) > 1`

Please run your code (e.g. with `-vv`) and analyze the output to ensure it is logical!  Clearly you did not and we can see gstreamer rejects 1:n pipelines.

Reply:
I have blocked in config spec to exclude 1:n tests and take only 1:1 tests as in future if 1:n tests enabled for gstreamer it will be easy to update config

Reply:
No, we need to handle it in the transcode test code.  I doubt 1:n will ever be supported in gstreamer using the pipeline syntax that you've implemented.

Reply:
ok, added check to skip 1ton tests